docs/sources/flow/reference/components/otelcol.processor.transform.md

```diff
@@ -0,0 +1,575 @@
+log_statements | [log_statements][] | Statements which transform logs. | no
+
+[trace_statements]: #trace_statements-block
+[metric_statements]: #metric_statements-block
+[log_statements]: #log_statements-block
+[output]: #output-block
+
+[OTTL Context]: #ottl-context
+
+### trace_statements block
+
+The `trace_statements` block specifies statements which transform trace telemetry signals. 
+Multiple `trace_statements` blocks can be specified.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`context` | `string` | OTTL Context to use when interpreting the associated statements. | | yes
+`statements` | `list(string)` | A list of OTTL statements. | | yes
+
+The supported values for `context` are:
+* `resource`: Use when interacting only with OTLP resources (for example, resource attributes).
+* `scope`: Use when interacting only with OTLP instrumentation scope (for example, the name of the instrumentation scope).
+* `span`: Use when interacting only with OTLP spans.
+* `spanevent`: Use when interacting only with OTLP span events.
+
+See [OTTL Context][] for more information about how ot use contexts.
+
+### metric_statements block
+
+The `metric_statements` block specifies statements which transform metric telemetry signals. 
+Multiple `metric_statements` blocks can be specified.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`context` | `string` | OTTL Context to use when interpreting the associated statements. | | yes
+`statements` | `list(string)` | A list of OTTL statements. | | yes
+
+The supported values for `context` are:
+* `resource`: Use when interacting only with OTLP resources (for example, resource attributes).
+* `scope`: Use when interacting only with OTLP instrumentation scope (for example, the name of the instrumentation scope).
+* `metric`: Use when interacting only with individual OTLP metrics.
+* `datapoint`: Use when interacting only with individual OTLP metric data points.
+
+Refer to [OTTL Context][] for more information about how to use contexts.
+
+### log_statements block
+
+The `log_statements` block specifies statements which transform log telemetry signals. 
+Multiple `log_statements` blocks can be specified.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`context` | `string` | OTTL Context to use when interpreting the associated statements. | | yes
+`statements` | `list(string)` | A list of OTTL statements. | | yes
+
+The supported values for `context` are:
+* `resource`: Use when interacting only with OTLP resources (for example, resource attributes).
+* `scope`: Use when interacting only with OTLP instrumentation scope (for example, the name of the instrumentation scope).
+* `log`: Use when interacting only with OTLP logs.
+
+See [OTTL Context][] for more information about how ot use contexts.
+
+### OTTL Context
```

Comment:
Consider putting this section in a shared .md if you think it will be needed on multiple components in the future.

Reply:
I think it's probably just going to be this component. If other components do need it, we could move it to a shared md at a later points when we do have that need.